src/main/java/net/mcreator/ui/views/editor/image/tool/tools/ColorPickerTool.java

```diff
@@ -22,24 +22,57 @@
 import net.mcreator.ui.init.UIRES;
 import net.mcreator.ui.views.editor.image.canvas.Canvas;
 import net.mcreator.ui.views.editor.image.tool.component.ColorSelector;
+import net.mcreator.ui.views.editor.image.tool.tools.event.ToolActivationEvent;
 import net.mcreator.ui.views.editor.image.versioning.VersionManager;
 
 import java.awt.*;
+import java.awt.event.MouseEvent;
 
 public class ColorPickerTool extends AbstractTool {
 
 	public ColorPickerTool(Canvas canvas, ColorSelector colorSelector, VersionManager versionManager) {
 		super("Color picker", "A tool that sets current foreground color to the color under the cursor",
-				UIRES.get("img_editor.picker"), canvas, colorSelector, versionManager);
-		noSettings(true);
+				UIRES.get("img_editor.picker"), canvas, colorSelector, versionManager, "Last picked color opacity:");
 	}
 
-	@Override public boolean process(ZoomedMouseEvent mouseEvent) {
-		if (layer.in(mouseEvent.getX(), mouseEvent.getY())) {
-			Color c = new Color(layer.getRGB(mouseEvent.getX() - layer.getX(), mouseEvent.getY() - layer.getY()));
-			colorSelector.setForegroundColor(c);
+	@Override public boolean process(ZoomedMouseEvent e) {
+		mouseMoved(e);
+		if (layer.in(e.getX(), e.getY())) {
+			Color c = new Color(layer.getRGB(e.getX() - layer.getX(), e.getY() - layer.getY()), true);
+			if (e.isShiftDown()) {
+				colorSelector.setBackgroundColor(new Color(c.getRGB()));
+			} else {
+				colorSelector.setForegroundColor(new Color(c.getRGB()));
+			}
+			opacitySlider.setValue(c.getAlpha());

```

Comment:
Again this should be tool specific. You can define tool panel elements in a streamlined way (check other components).

Reply:
:heavy_check_mark: Fixed